### ISSUE TITLE
TypeError: Path must be a string. Received undefined

### ISSUE THREAD

**Issue by XhmikosR** [postcss/autoprefixer#865]
Hi, there.

I'm using gulp-postcss with a few plugins like this:

```js
'use strict';

var gulp = require('gulp');
var postcss = require('gulp-postcss');
var sass = require('gulp-sass');
var plumber = require('gulp-plumber');
var gutil = require('gulp-util');
var autoprefixer = require('autoprefixer');
var mqpacker = require('css-mqpacker');
var combineDupeSelectors = require('postcss-combine-duplicated-selectors');
var zIndex = require('postcss-zindex');
var browsersync = require('browser-sync');
var config = require('../../config');

var postcssProcessors = [
    autoprefixer(),
    mqpacker(config.styles.options.mqpacker),
    zIndex(),
    combineDupeSelectors()
];

function onError (err) {
    gutil.beep();
    console.log(err);
    this.emit('end');
}

gulp.task('styles', function() {
    browsersync.notify('Compiling CSS files');

    return gulp.src(config.styles.src)
        .pipe(plumber({
            errorHandler: onError
        }))
        .pipe(sass(config.sass.options))
        .pipe(postcss(postcssProcessors))
        .pipe(gulp.dest(config.styles.dest))
        .pipe(browsersync.stream());
});
```

I have a `.broswerslistrc` file next to my package.json in my project's root folder.

```
C:\Users\xmr\Desktop\foobar>gulp
[14:21:36] Using gulpfile ~\Desktop\foobar\gulpfile.js
[14:21:36] Starting 'styles'...
{ TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:207:7)
    at eachParent (C:\Users\xmr\Desktop\foobar\node_modules\browserslist\index.js:65:18)
    at getStat (C:\Users\xmr\Desktop\foobar\node_modules\browserslist\index.js:79:12)
    at browserslist (C:\Users\xmr\Desktop\foobar\node_modules\browserslist\index.js:188:15)
    at Browsers.parse (C:\Users\xmr\Desktop\foobar\node_modules\autoprefixer\lib\browsers.js:61:16)
    at new Browsers (C:\Users\xmr\Desktop\foobar\node_modules\autoprefixer\lib\browsers.js:52:30)
    at Supports.prefixer (C:\Users\xmr\Desktop\foobar\node_modules\autoprefixer\lib\supports.js:48:24)
    at Supports.prefixed (C:\Users\xmr\Desktop\foobar\node_modules\autoprefixer\lib\supports.js:93:29)
    at C:\Users\xmr\Desktop\foobar\node_modules\autoprefixer\lib\supports.js:316:39
  name: 'TypeError',
  message: 'Path must be a string. Received undefined',
  stack: 'TypeError: Path must be a string. Received undefined\n    at assertPath (path.js:28:11)\n
   at Object.resolve (path.js:207:7)\n    at eachParent (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\browserslist\\index.js:65:18)\n    at getStat (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\browserslist\\index.js:79:12)\n    at browserslist (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\browserslist\\index.js:188:15)\n    at Browsers.parse (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\autoprefixer\\lib\\browsers.js:61:16)\n    at new Browsers (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\autoprefixer\\lib\\browsers.js:52:30)\n    at Supports.prefixer (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\autoprefixer\\lib\\supports.js:48:24)\n    at Supports.prefixed (C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\autoprefixer\\lib\\supports.js:93:29)\n
 at C:\\Users\\xmr\\Desktop\\foobar\\node_modules\\autoprefixer\\lib\\supports.js:316:39',
  fileName: 'C:\\Users\\xmr\\Desktop\\foobar\\sass\\main.css',
  showStack: true,
  showProperties: true,
  plugin: 'gulp-postcss' }

```

```
C:\Users\xmr\Desktop\foobar>npm ls autoprefixer
foobar@1.0.0 C:\Users\xmr\Desktop\foobar
+-- autoprefixer@7.1.2
`-- stylelint@7.13.0
  `-- autoprefixer@6.7.7

C:\Users\xmr\Desktop\foobar>npm ls browserslist
foobar@1.0.0 C:\Users\xmr\Desktop\foobar
+-- autoprefixer@7.1.2
| `-- browserslist@2.2.1
`-- stylelint@7.13.0
  +-- autoprefixer@6.7.7
  | `-- browserslist@1.7.7
  +-- doiuse@2.6.0
  | `-- browserslist@1.7.7
  `-- stylehacks@2.3.2
    `-- browserslist@1.7.7
```

I'm using node.js 8.2.1 on Windows 7 64-bit.

If I go back to my previous deps it works fine:

```
C:\Users\xmr\Desktop\foobar>npm ls autoprefixer
foobar@1.0.0 C:\Users\xmr\Desktop\foobar
+-- autoprefixer@7.1.2
`-- stylelint@7.13.0
  `-- autoprefixer@6.7.7


C:\Users\xmr\Desktop\foobar>npm ls browserslist
foobar@1.0.0 C:\Users\xmr\Desktop\foobar
+-- autoprefixer@7.1.2
| `-- browserslist@2.1.5
`-- stylelint@7.13.0
  +-- autoprefixer@6.7.7
  | `-- browserslist@1.7.7
  +-- doiuse@2.6.0
  | `-- browserslist@1.7.7
  `-- stylehacks@2.3.2
    `-- browserslist@1.7.7
```

So the culprit as you can see from the above seems to be `browserslist@2.2.1`.

Has a breaking change been introduced and I need to change something on my side? Initially I saw the error when I tried to update stylelint, but then I managed to get the same error by just clearing my node_modules and doing `npm i` again which picks up the aforementioned package versions.

Any ideas are welcome!

**Comment by XhmikosR** [comment]
Closing this since it's a browserslist issue https://github.com/ai/browserslist/issues/158

**Comment by ai** [comment]
Fixed in Browserslist 2.2.2
